src/app-layout/__tests__/trigger-button.test.tsx

```diff
@@ -0,0 +1,353 @@
+import createWrapper from '../../../lib/components/test-utils/dom';
+
+import visualRefreshStyles from '../../../lib/components/app-layout/visual-refresh/styles.css.js';
+import toolbarTriggerButtonStyles from '../../../lib/components/app-layout/visual-refresh-toolbar/toolbar/trigger-button/styles.css.js';
+
+const MockUseAppLayoutInternals = jest.spyOn(AllContext, 'useAppLayoutInternals').mockName('useAppLayoutInternals');
+
+const mockDrawerId = 'mock-drawer-id';
+const mockTestId = `awsui-app-layout-trigger-${mockDrawerId}`;
+
+const mockProps = {
+  ariaLabel: 'Aria label',
+  className: 'class-from-props',
+  iconName: 'bug',
+  testId: mockTestId,
+};
+const mockOtherEl = {
+  class: 'other-el-class',
+  text: 'other-element',
+};
+const mockUseLayoutInternalValues = {
+  hasOpenDrawer: false,
+  isMobile: false,
+};
+
+const mockEventBubble = {
+  bubbles: true,
+  isTrusted: true,
+  relatedTarget: null,
+};
+
+const mockEventBubbleWithShiftFocus = {
+  ...mockEventBubble,
+  relatedTarget: {
+    dataset: {
+      shiftFocus: 'last-opened-toolbar-trigger-button',
+    },
+  },
+};
+
+const renderVisualRefreshTriggerButton = (
+  props: Partial<VisualRefreshTriggerButtonProps> = {},
+  useAppLayoutInternalsValues: Partial<AllContext.AppLayoutInternals> = {},
+  ref: React.Ref<ButtonProps.Ref>
+) => {
+  MockUseAppLayoutInternals.mockReturnValue({
+    ...mockUseLayoutInternalValues,
+    ...useAppLayoutInternalsValues,
+  } as AllContext.AppLayoutInternals);
+  const renderProps = { ...mockProps, ...props };
+  const { container, rerender, getByTestId, getByText } = render(
+    <div>
+      <VisualRefreshTriggerButton {...(renderProps as VisualRefreshTriggerButtonProps)} ref={ref} />
+      <span className={mockOtherEl.class}>{mockOtherEl.text}</span>
+    </div>
+  );
+  const wrapper = createWrapper(container).findByClassName(visualRefreshStyles['trigger-wrapper'])!;
+  return { wrapper, rerender, getByTestId, getByText };
+};
+
+const renderVisualRefreshToolbarTriggerButton = (
+  props: Partial<VisualRefreshToolbarTriggerButtonProps> = {},
+  ref: React.Ref<ButtonProps.Ref>
+) => {
+  const renderProps = { ...mockProps, ...props };
+  const { container, rerender, getByTestId, getByText } = render(
+    <div>
+      <VisualRefreshToolbarTriggerButton {...(renderProps as VisualRefreshToolbarTriggerButtonProps)} ref={ref} />
+      <span className={mockOtherEl.class}>{mockOtherEl.text}</span>
+    </div>
+  );
+  const wrapper = createWrapper(container).findByClassName(toolbarTriggerButtonStyles['trigger-wrapper'])!;
+  return { wrapper, rerender, getByTestId, getByText };
+};
+
+describe('Visual refresh trigger-button (not in appLayoutWidget toolbar)', () => {
+  beforeEach(() => jest.clearAllMocks());
+
+  //tests for both desktop and mobile
+  describe.each([true, false])('Toolbar trigger-button with isMobile=%s', isMobile => {
+    //tests for both drawer being opend and closed
```

Comment:
This as well - the variables isMobile and hasOpenDrawer convey enough info, I think.